### PR TITLE
Authfeat

### PR DESCRIPTION
**Pull request by markivancho**

### CODE REVIEW

File: app/controllers/users/omniauth_callbacks_controller.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+class Users::OmniauthCallbacksController < Devise::OmniauthCallbacksController
+  def google_oauth2
+      # You need to implement the method below in your model (e.g. app/models/user.rb)
```
Comment (AndrewMatsievsky) [inline app/controllers/users/omniauth_callbacks_controller.rb]: @markivancho remove default comments, please

File: app/controllers/users/omniauth_callbacks_controller.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+      @user = User.from_omniauth(request.env['omniauth.auth'])
+
+      if @user.persisted?
+        flash[:notice] = I18n.t 'devise.omniauth_callbacks.success', kind: 'Google'
+        sign_in_and_redirect @user, event: :authentication
+      else
+        session['devise.google_data'] = request.env['omniauth.auth'].except(:extra) # Removing extra as it can overflow some session stores
```
Comment (AndrewMatsievsky) [inline app/controllers/users/omniauth_callbacks_controller.rb]: same here

File: app/models/user.rb [hunk 3]
```diff
@@ -2,5 +2,20 @@ class User < ApplicationRecord
   # Include default devise modules. Others available are:
```
Comment (AndrewMatsievsky) [inline app/models/user.rb]: comment

File: app/models/user.rb [hunk 4]
```diff
@@ -2,5 +2,20 @@ class User < ApplicationRecord
...
-         :recoverable, :rememberable, :trackable, :validatable
-end
+         :recoverable, :rememberable, :trackable, :validatable,
+         :omniauthable, omniauth_providers: [:google_oauth2]
+
+  def self.from_omniauth(access_token)
+    data = access_token.info
+    user = User.where(email: data['email']).first
+
+    # Uncomment the section below if you want users to be created if they don't exist
```
Comment (AndrewMatsievsky) [inline app/models/user.rb]: and here one more

File: db/migrate/20170802195928_add_omniauth_to_users.rb [hunk 5]
```diff
@@ -0,0 +1,6 @@
+class AddOmniauthToUsers < ActiveRecord::Migration[5.1]
+  def change
```
Comment (jizak) [inline db/migrate/20170802195928_add_omniauth_to_users.rb:2]: I think we have to many columns here.

**Review by jizak — APPROVED** [review]

**Pull request merged**